addon/href-to.js

```diff
@@ -60,7 +58,7 @@ export default class {
     let id = this.target.id;
     if (id) {
       let componentInstance = this.applicationInstance.lookup('-view-registry:main')[id];
-      isLinkComponent = componentInstance && componentInstance instanceof LinkComponent;
+      isLinkComponent = componentInstance && componentInstance.constructor.superclass.toString() === '@ember/routing/link-component';
```

Comment:
I actually mentioned that in the PRs description:
> LinkComponent defines a toString() that outputs @ember/routing/link-component since version 3.1.0 (released on 13 Apr 2018): [emberjs/ember.js@v3.1.0/packages/ember-glimmer/lib/components/link-to.ts#L825](https://github.com/emberjs/ember.js/blob/v3.1.0/packages/ember-glimmer/lib/components/link-to.ts?rgh-link-date=2023-02-10T01%3A48%3A31Z#L825)

So, unless we plan to support older versions than that, yes.

I see that in ember 3.0.0, `toString()` output the string `'LinkComponent'`: https://github.com/emberjs/ember.js/blob/v3.0.0/packages/ember-glimmer/lib/components/link-to.ts#L825

I was able to track this `'LinkComponent'` back to version 1.x: https://github.com/emberjs/ember.js/blob/v1.13.13/packages/ember-routing-views/lib/views/link.js#L459
I think it's pretty safe to assume "since forever".

So, I can add `|| componentInstance.constructor.superclass.toString() === 'LinkComponent'` to cover older versions, if you feel necessary.